src/test/java/spoon/test/type/TypeTest.java

```diff
@@ -0,0 +1,115 @@
+ * FITNESS FOR A PARTICULAR PURPOSE. See the CeCILL-C License for more details.
+ *
+ * The fact that you are presently reading this means that you have had
+ * knowledge of the CeCILL-C license and that you accept its terms.
+ */
+
+package spoon.test.type;
+
+import org.junit.Test;
+import spoon.Launcher;
+import spoon.reflect.code.BinaryOperatorKind;
+import spoon.reflect.code.CtBinaryOperator;
+import spoon.reflect.code.CtBlock;
+import spoon.reflect.code.CtFieldRead;
+import spoon.reflect.code.CtLocalVariable;
+import spoon.reflect.code.CtTypeAccess;
+import spoon.reflect.declaration.CtAnnotation;
+import spoon.reflect.declaration.CtClass;
+import spoon.reflect.declaration.CtMethod;
+import spoon.reflect.factory.Factory;
+import spoon.reflect.visitor.filter.TypeFilter;
+import spoon.test.TestUtils;
+import spoon.test.type.testclasses.Pozole;
+
+import java.util.List;
+
+import static junit.framework.TestCase.assertFalse;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class TypeTest {
+	@Test
+	public void testTypeAccessForDotClass() throws Exception {
+		// contract: When we use .class on a type, this must be a CtTypeAccess.
+		final String target = "./target/type";
+		final Launcher launcher = new Launcher();
+		launcher.addInputResource("./src/test/java/spoon/test/type/testclasses");
+		launcher.setSourceOutputDirectory(target);
+		launcher.getEnvironment().setNoClasspath(true);
+		launcher.run();
+
+		final CtClass<Pozole> aPozole = launcher.getFactory().Class().get(Pozole.class);
+		final CtMethod<?> make = aPozole.getMethodsByName("make").get(0);
+
+		final List<CtFieldRead<?>> fieldClasses = make.getElements(new TypeFilter<CtFieldRead<?>>(CtFieldRead.class) {
+			@Override
+			public boolean matches(CtFieldRead<?> element) {
+				return "class".equals(element.getVariable().getSimpleName()) && super.matches(element);
+			}
+		});
+		assertEquals(4, fieldClasses.size());
+		for (CtFieldRead<?> fieldClass : fieldClasses) {
+			assertTrue(fieldClass.getTarget() instanceof CtTypeAccess);
+		}
+
+		TestUtils.canBeBuilt(target, 8, true);
+	}
+
+	@Test
+	public void testTypeAccessOnPrimitive() throws Exception {
+		Factory factory = TestUtils.createFactory();
+		CtClass<?> clazz = factory.Code().createCodeSnippetStatement( //
+				"class X {" //
+						+ "public void foo() {" //
+						+ " Class klass=null;" //
+						+ "  boolean x= (klass == short.class);" //
+						+ "}};").compile();
+		CtMethod<?> foo = (CtMethod<?>) clazz.getMethods().toArray()[0];
+
+		CtBlock<?> body = foo.getBody();
+		CtLocalVariable<?> ass = body.getStatement(1);
+		CtBinaryOperator<?> op = (CtBinaryOperator<?>) ass.getDefaultExpression();
+		assertEquals("Class", op.getLeftHandOperand().getType().getSimpleName());
+		assertFalse(op.getLeftHandOperand().getType().isPrimitive());
+		assertEquals("short", op.getRightHandOperand().getType().getSimpleName());
```

Comment:
Should return `Class` and `op.getLeftHandOperand().getTarget().getSimpleName()` return `short`.